File: AtCoder/AtCoder Beginner Contest/049/C/C/ABC_049_C_02.c
#include <stdio.h>
#include <regex.h>

int main()
{
    char str[1024000];
    scanf("%s", str);

    const char regex[] = "^(dream|dreamer|erase|eraser)*$";
    regex_t regexBuffer;

    if (regcomp(&regexBuffer, regex, REG_EXTENDED | REG_NEWLINE) != 0) {
        return -1;
    }

    if (regexec(&regexBuffer, str, 0, NULL, 0) == 0) {
        printf("YES");
    } else {
        printf("NO");
    }

    regfree(&regexBuffer);
    return 0;
}

File: AtCoder/AtCoder Beginner Contest/049/C/Go/ABC_049_C_02.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	var S string
	fmt.Scanf("%s", &S)

	var len_old, len_new int
	for {
		len_old = len(S)
		for _, v := range []string{"eraser", "erase", "dreamer", "dream"} {
			S = strings.TrimSuffix(S, v)
		}
		len_new = len(S)
		if len_old - len_new == 0 {break}
	}

	if len(S) == 0 {
		fmt.Printf("YES")
	} else {
		fmt.Printf("NO")
	}
}